### PR TITLE
Corrected LearnerPrediction and Prediction tests for Supervised Tests…

### PR DESCRIPTION
**Pull request by rold2007**
…. They were failing once in a while.

Corrected exception when the test samples don't contain all potential labels.
Corrected typos.

### CODE REVIEW

**Comment by rold2007** [comment]
I was trying to fix an issue in NaiveBayesGenerator and I noticed that some tests were failing once in a while. It made it hard to figure out if it was my changes or something else so I decided to fix the fails separately.

Here is a longer description of the changes I made. Note that there were many ways to prevent the fails. Let me know if you would have preferred another approach.

**LearnerPrediction and Prediction fails**: This was mainly caused by the fact that the test datasets are quite small. Since the train/test samples are selected randomly we were sometimes _unlucky_ and the accuracy was not satisfying. Considering that by using a 0.8 ratio of train vs test samples on a 14 samples dataset, this leaves only 3 test samples. The current minimum accuracy was 0.75 which would not allow even one fail. I reduced this to 0.66 and increased the _repeat_ from 10 to 100.
Another recurring fail was that the _best_ model couldn't always predict the hardcoded sample for Iris even when the model accuracy was high. Looking at the [scatterplot](https://en.wikipedia.org/wiki/Iris_flower_data_set#/media/File:Iris_dataset_scatterplot.svg) for Iris I noticed that SepalLength and SepalWidth were a little bit off from what we would expect of Iris-Setosa so I adjusted these values.

**Index out of range exception**: The GenerateModel() method in Learner.cs was extracting the potential labels from the test set. In a context of low test samples it often happened that the test set didn't even include all the trained labels. When the model would (_wrongfully_) predict a label which was not in the test set it would lead to an invalid descriptor label index. I added a parameter to prevent updating the labels list. Another way to repair this could have been to clone the descriptor from the train set but I found no clone method for Descriptor.

Note that some numl tests are still failing but they are unrelated to my changes. I might give them a look once I fixed NaiveBayesGenerator...

File: Src/numl/Model/Descriptor.cs [hunk 1]
```diff
@@ -193,13 +193,13 @@ public IEnumerable<double> Convert(object item, bool withLabel)
 		/// <param name="items">Examples.</param>
 		/// <param name="withLabels">True to include labels, otherwise False</param>
 		/// <returns>Lazy double enumerable of doubles.</returns>
-		public IEnumerable<IEnumerable<double>> Convert(IEnumerable<object> items, bool withLabels = true)
+		public IEnumerable<IEnumerable<double>> Convert(IEnumerable<object> items, bool withLabels = true, bool updateLabels = true)
```
Comment (sethjuarez) [inline Src/numl/Model/Descriptor.cs]: What does update labels do?

Comment (sethjuarez) [inline Src/numl/Model/Descriptor.cs:196]: I SEE! It is for labels that have never been seen. GENIUS I SAY!

**Review by sethjuarez — COMMENTED** [review]
So what happens when new labels come in?

File: Src/numl/Model/Descriptor.cs (hunk 1 above)
Comment (rold2007) [reply Src/numl/Model/Descriptor.cs:196]: Actually I would say it is for labels that have _already_ been seen. This parameter is used to support my modification in Learner.cs. When GenerateModel() is testing a model it needs to use the labels from the training set (already seen). Previously it was using the labels present only in the test set, which could be very limited compared to the training set. But the trained model could still predict a label not present in the test set, causing and exception in GenerateModel().
Was my explanation clear enough ?

**Comment by rold2007** [comment]
Hey @sethjuarez , I would like to submit another pull request but it is dependent on this one. Is it possible for you to merge it ? Is there any improvement you would need me to do first ?

Thanks
